### ISSUE TITLE
Android Paging library 2 support for heterogeneous models

### ISSUE THREAD

**Issue by aleonq** [airbnb/epoxy#1078]
The PagedListEpoxyController only has support for one type of model that can be paginated. This doesn't work in the cases like:

1. Static views on top + paged list at bottom
2. Horizontally scrollable paged content of multiple types with of with/without vertically scrollable paged content at the bottom.

Case 1:
==================
Name: **Darth**
Current location: **X-34-34Z**
-------------------------
_Location History_

1. Planet Soros
2. Planet Merus
3. .
4. .
5. .
6. .
7. .

==================

Case 2:
==================
**Cart:**

1. Apple Pie
2. Fruit Cake

-------------------------
_Chocolates:_

| Dark          | Marz | Fruits   |  ***  |
| Chocolate | Bar.   | & Nuts |  ***  |           <-----scroll direction------->

.
.
.
.
.

==================

@elihart , This sounds more like a feature request than an issue. Sorry about creating the issue. Please let me know if there is another way to create a feature request.
I could not find docs/sample regarding android paging(v2) support other than this way. Please let mw know if there is already a way.

Thanks, take care!

**Comment by elihart** [comment]
Yes, this would be a feature request. It is a bit hard to support as we would need a new API to control caching multiple item types and I haven't seen a good way to approach it yet.

If you need multiple item types, there is basic support via overriding `addModels` to insert additional models as needed. One improvement might be to enable `buildItemModel` to return a list of models, but a difficult there is it makes it hard to know when we need to load more items in the paged list.

Unless someone is interested in solving these problems in a clean way I'd recommend sticking to your own custom EpoxyController where you manage model building yourself (the downside is just that you lose some performance from caching the built models automatically)

**Comment by aleonq** [comment]
Thanks @elihart for prompt response. Appreciate it.

I saw that there is paging 2 integration support [here PagingEpoxyController](https://github.com/airbnb/epoxy/blob/cfa2bfae32a6cd4871f0dc1e3817474c60b59d08/epoxy-paging/src/main/java/com/airbnb/epoxy/paging/PagingEpoxyController.java#L3-L5)

Were you talking in context of this class?
Can you please point me to anything that might be useful in utilizing Paging 2 (since 3 is in alpha) library?
Reason I dont want to use bare Epoxy controller is that the I wont be able to use Paging library with it. If I were to use the class PagingEpoxyController, I might end up wrapping that in a nested scrolling view with my header view^, which would make RecyclerView get and bind all available items. Thoughts on how would you approach this?

**Comment by rohithThammaiah** [comment]
A short-term fix for anyone still wondering about this is to use a `EpoxyModelGroup` with a layout having a single `LinearLayout` with orientation set to vertical and return the model group in `PagingEpoxyController.buildItemModel`. 
@elihart Let me know if this a good or a bad approach because I don't think Sticky headers for views inside the model group will work.
